pkg/core/blockchain.go

```diff
@@ -401,71 +401,45 @@ func (bc *Blockchain) persistBlock(block *Block) error {
 	if err := assets.commit(batch); err != nil {
 		return err
 	}
-	if err := bc.PutBatch(batch); err != nil {
+	if err := bc.memStore.PutBatch(batch); err != nil {
 		return err
 	}
 
-	bc.persistedHeight = block.Index
+	atomic.StoreUint32(&bc.blockHeight, block.Index)
 	return nil
 }
 
-// persist flushed current block cache to the persistent storage.
-func (bc *Blockchain) persist(ctx context.Context) (err error) {
+// persist flushed current in-memory store to the persistent storage.
```

Comment:
nit: dunno if it's correct to say:
"flushed current in-memory store to the persistent storage"
I would probably say:
"flushed current in-memory store data to the persistent storage"